crates/proto/src/xfer/dns_request.rs

```diff
@@ -54,6 +56,34 @@ pub struct DnsRequest {
 }
 
 impl DnsRequest {
+    #[cfg(feature = "std")]
+    pub(crate) fn from_query(mut query: Query, options: DnsRequestOptions) -> Self {
+        // build the message
+        let mut message = Message::query();
+        let mut original_query = None;
+
+        if options.case_randomization {
+            original_query = Some(query.clone());
+            query.name.randomize_label_case();
+        }
+
+        message
+            .add_query(query)
+            .set_recursion_desired(options.recursion_desired);
+
+        // Extended dns
+        if options.use_edns {
+            message
+                .extensions_mut()
+                .get_or_insert_with(Edns::new)
+                .set_max_payload(MAX_PAYLOAD_LEN)
+                .set_version(0)
```

Comment:
The default version is `0` so I would be tempted to remove the explicit `set_version()` call here.